File: app/src/main/java/com/example/navigationdrawerapp/DividerItemDecoration.java
package com.example.navigationdrawerapp;

import android.graphics.Canvas;
import android.graphics.Rect;
import android.graphics.drawable.ColorDrawable;
import android.view.View;

import androidx.annotation.NonNull;
import androidx.recyclerview.widget.RecyclerView;

public class DividerItemDecoration extends RecyclerView.ItemDecoration{

    private static final String TAG = DividerItemDecoration.class.getSimpleName();

    private final ColorDrawable divider;
    private final int height;

    public DividerItemDecoration(int resId, int height) {
        this.divider = new ColorDrawable(resId);
        this.height = height;
    }

    @Override
    public void onDraw(@NonNull Canvas canvas, @NonNull RecyclerView parent, @NonNull RecyclerView.State state) {
        super.onDraw(canvas, parent, state);
        final int left = parent.getPaddingLeft();
        final int right = parent.getWidth() - parent.getPaddingRight();

        final int childCount = parent.getChildCount();
        for (int index = 0; index < childCount; index++) {
            final View child = parent.getChildAt(index);
            final RecyclerView.LayoutParams params = (RecyclerView.LayoutParams) child.getLayoutParams();
            final int top = child.getBottom() + params.bottomMargin;
            final int bottom = top + divider.getIntrinsicHeight();
            //if (child.getVisibility() == View.VISIBLE && child.getId() != -1) {
                divider.setBounds(left, top, right, bottom);
                divider.draw(canvas);
            //}
        }
    }

    @Override
    public void getItemOffsets(@NonNull Rect outRect, @NonNull View view, @NonNull RecyclerView parent, @NonNull RecyclerView.State state) {
        super.getItemOffsets(outRect, view, parent, state);
        outRect.bottom = height;
    }
}

File: app/src/main/java/com/example/navigationdrawerapp/DividerItemDecoration.kt
package com.example.navigationdrawerapp

import android.graphics.Canvas
import android.graphics.Rect
import android.graphics.drawable.ColorDrawable
import android.view.View
import androidx.recyclerview.widget.RecyclerView

class DividerItemDecoration : RecyclerView.ItemDecoration {


    companion object {
        private val TAG = DividerItemDecoration::class.java.getSimpleName()
    }

    private val divider : ColorDrawable
    private val height : Int

    constructor(resId : Int, height : Int) {
        divider = ColorDrawable(resId)
        this.height = height
    }

    override fun onDraw(canvas : Canvas, parent : RecyclerView, state : RecyclerView.State) {
        super.onDraw(canvas, parent, state)

        val left = parent.paddingLeft
        val right = parent.width - parent.paddingRight

        val childCount = parent.childCount
        for (i in 0 until childCount) {
            val child : View = parent.getChildAt(i)
            val params = child.getLayoutParams() as RecyclerView.LayoutParams
            val top: Int = child.getBottom() + params.bottomMargin
            val bottom = top + divider!!.intrinsicHeight
            divider.setBounds(left, top, right, bottom)
            divider.draw(canvas)
        }

    }

    override fun getItemOffsets(outRect : Rect, view : View, parent : RecyclerView, state : RecyclerView.State) {
        super.getItemOffsets(outRect, view, parent, state)
        outRect.bottom = height
    }
}